Heim/Presentation/Presentation/Music/View/MusicMatchViewController.swift

```diff
@@ -0,0 +1,184 @@
+//
+//  MusicMatchViewController.swift
+//  Presentation
+//
+//  Created by 김미래 on 11/25/24.
+//
+
+import UIKit
+
+// TODO: 수정
+struct Music {
+  let title: String
+  let artist: String
+}
+
+final class MusicMatchViewController: UIViewController, Coordinatable {
+  // MARK: - Properties
+  // TODO: let 수정
+  private var musicDataSources: [Music]
+  weak var coordinator: DefaultMusicMatchCoordinator?
+
+  // MARK: - UI Components
+  private let backgroundImageView: UIImageView = {
+    let imageView = UIImageView()
+    imageView.image = .background
+    imageView.contentMode = .scaleAspectFill
+    return imageView
+  }()
+
+  private let titleLabel: CommonLabel = {
+    let label = CommonLabel(font: .bold, size: LayoutConstants.titleThree)
+    label.text = "하임이가 추천하는 음악을 가져왔어요!"
+    return label
+  }()
+
+  private let musicTableView: UITableView = {
+    let tableView = UITableView(frame: .zero)
+    tableView.registerCellClass(cellType: MusicTableViewCell.self)
+    tableView.separatorStyle = .none
+    tableView.isScrollEnabled = false
+    tableView.layer.masksToBounds = true
+    tableView.layer.cornerRadius = LayoutConstants.cornerRadius
+    tableView.isScrollEnabled = true
+    return tableView
+  }()
+
+  private let homeButton: CommonRectangleButton = {
+    let button = CommonRectangleButton(
+      fontStyle: .boldFont(ofSize: LayoutConstants.homeButtonFont),
+      backgroundColor: .primary,
+      radius: 10
+    )
+    button.setTitle("메인 화면으로 이동하기",
+                    for: .normal)
+    return button
+  }()
+
+  // MARK: - Initializer
+  init(musics: [Music]) {
+    self.musicDataSources = musics
+    // TODO: 삭제
+    self.musicDataSources = [Music(title: "슈퍼노바", artist: "#감성힙합#플레이리스트 #해시태그 #해시태..."),
+                             Music(title: "슈퍼노바", artist: "#감성힙합#플레이리스트 #해시태그 #해시태..."),
+                             Music(title: "슈퍼노바", artist: "#감성힙합#플레이리스트 #해시태그 #해시태..."),
+                             Music(title: "슈퍼노바", artist: "#감성힙합#플레이리스트 #해시태그 #해시태..."),
+                             Music(title: "슈퍼노바", artist: "#감성힙합#플레이리스트 #해시태그 #해시태...")]
+    super.init(nibName: nil, bundle: nil)
+  }
+
+  required init?(coder: NSCoder) {
+    fatalError("init(coder:) has not been implemented")
+  }
+
+  // MARK: - LifeCycle
+  override func viewDidLoad() {
+    setupViews()
+    setupLayoutConstraints()
+  }
+
+  override func viewDidLayoutSubviews() {
+    setupTableViewGradient()
+  }
+
+  @objc func homeButtondidTap() {
+    coordinator?.pushHomeView()
+  }
+}
+
+extension MusicMatchViewController: UITableViewDelegate, UITableViewDataSource {
```

Comment:
tableview를 다루는 datasources(테이블뷰의 데이터 관리 로직)를 별도의 클래스로 분리하는 것은 어떨까요??
```suggestion
extension MusicMatchViewController: UITableViewDelegate, UITableViewDataSource {
```

Reply:
네 그게 더 깔끔해 보이네요 ! 수정했습니다!